core/controllers/reader.py

```diff
@@ -281,12 +282,22 @@ def get(self, exploration_id):
                     'data_schema_version': data_schema_version
                 }
 
+        pretest_questions = (
+            story_services.get_pretests_for_exploration(exploration_id))
+
+        pretest_question_dicts = []
+        if pretest_questions is not None:
```

Comment:
This should never happen -- worst case, it should be an empty list.

Reply:
Oh, I see -- you're using None as sentinel for "it's not in a story". Seems a bit indirect/unexpected.

Here's an idea: if you look at how "exploration in collection" is detected, we use a URL parameter. Why don't we do that instead here? Then we can take the story ID from the passed-in params and verify that the exploration ID is in it. And then we can retrieve pre-test questions only in the case where we're told (and we confirm) that the exploration is in a story. Otherwise we raise an exception.

For the case when an exploration is not attached to a story I think returning an empty list is fine.

That way you have three outcomes:
- invalid params: 400 error
- exploration is in story: question list is populated
- exploration is not in any story: question list is empty

Reply:
Yeah, that seems fine. So, in this case, if a learner is playing the exploration from outside the context of a story (maybe, directly from library), then pretests won't be surfaced, right?

Reply:
Done, added a query parameter for story_id.